app/src/main/java/com/mborowiec/antennapod/activity/MainActivity.java

```diff
@@ -490,6 +495,22 @@ private void handleNavIntent() {
     protected void onNewIntent(Intent intent) {
         super.onNewIntent(intent);
         setIntent(intent);
+
+        handleIntent(intent);
+    }
+
+    /**
+     * Checks whether the activity was launched via a deep link.
+     *
+     * @param intent incoming intent
+     */
+    private void handleIntent(Intent intent) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            if (Objects.equals(intent.getAction(), Intent.ACTION_VIEW)) {
```

Comment:
Is there a guidline/reason/preference for Objects.equals(... compared to myInstance.equals(...?

Reply:
As far as I'm aware Objects.equals() is "null safe"